File: api/src/main/java/gg/mineral/api/plugin/event/player/PlayerJoinEvent.java
package gg.mineral.api.plugin.event.player;

import gg.mineral.api.entity.living.human.Player;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PlayerJoinEvent extends PlayerEvent {
    private double x, y, z;

    public PlayerJoinEvent(Player player) {
        super(player);
        this.x = player.getX();
        this.y = player.getY();
        this.z = player.getZ();
    }
}

File: api/src/main/kotlin/gg/mineral/api/plugin/event/player/PlayerJoinEvent.kt
package gg.mineral.api.plugin.event.player

import gg.mineral.api.entity.living.human.Player

class PlayerJoinEvent(
    player: Player,
    var x: Double = player.x,
    var y: Double = player.y,
    var z: Double = player.z,
    var yaw: Float = player.yaw,
    var pitch: Float = player.pitch
) :
    PlayerEvent(player)
